### PR TITLE
refactor: replace usage of verification funcs in 03-connection

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Replacing usage of:
- `VerifyClientState`
- `VerifyConsensusState`
- `VerifyConnection`
- `VerifyChannel`
- `VerifyPacketCommitment`
- `VerifyAcknowledgement`
- `VerifyPacketReceipt`
- `VerifyPacketReceiptAbsence`

with the generic `VerifyMembership` and `VerifyNonMembership` functions. 

closes: #1293 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by AdityaSripal — APPROVED** [review]
Nice work @damiannolan !

File: modules/core/03-connection/keeper/verify.go [hunk 1]
```diff
@@ -31,9 +35,27 @@ func (k Keeper) VerifyClientState(
...
+	}
+
+	bz, err := k.cdc.MarshalInterface(clientState)
+	if err != nil {
+		return err
+	}
+
+	if err := targetClient.VerifyMembership(
+		ctx, clientStore, k.cdc, height,
+		0, 0,
```
[the remark this answers is not in the document]
Comment (colin-axner) [reply modules/core/03-connection/keeper/verify.go:56]: Do you think we should add test cases for each of the verify functions with the time/block delay periods set to 0. That is, add test cases which will normally fail on the delay period not being passed (if 0, 0, isn't passed in). Thoughts?

**Review by colin-axner — APPROVED** [review]
Great work!

See comments regarding time/block delay. Since we are hard coding `0, 0`, I think it might be worth adding some more code docs/test cases to ensure it doesn't get accidentally changed in the future

File: modules/core/03-connection/keeper/verify.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/core/03-connection/keeper/verify.go:56]: Great idea, I'll add the inline comment. 

> add test cases which will normally fail on the delay period not being passed (if 0, 0, isn't passed in)

So if 0, 0, isn't passed in here it won't fail? there's nothing at the lightclient level which would return an error, and it needs to be generic to support both zero values and non zero values for delay periods, right?

[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/core/03-connection/keeper/verify.go]: Let's address this with a follow up if needs be.

**Pull request merged**